### ISSUE TITLE
Potentially add even more basic examples?

### ISSUE THREAD

**Issue by andrew-kennedy** [cginternals/globjects#336]
I've been trying to use this library for a day or two now, and while I'm new to openGL, I'm struggling to use this library to even draw a simple triangle, starting with my vertex data and color data in 2 separate `std::vector<glm::vec3>`'s. I can accomplish this with the normal openGL calls, but despite going through example after example, I can't seem to figure out how to translate the simple task of drawing one triangle to globjects code. Having even just one slightly more fleshed out case of regular openGL vs globjects would be super helpful. It's really hard to find documentation on this library elsewhere on the web.

**Comment by scheibel** [comment]
You're right, the current entry level of knowledge includes actual OpenGL knowledge of all object types and their relations. We should focus on easing usage, e.g., by providing more examples that show more aspects from OpenGL and how they are used with globjects.

As a long-term goal we want to provide a large amount of cool OpenGL demos and examples based on globjects but our programming capacity is currently low and we must focus on other projects right now.
I think a quick view into some of our projects based on globjects may help:
* https://github.com/hpicgs/glexamples
* https://github.com/cgcostume/multiframesampling
* https://github.com/cginternals/glhimmel

In addition, you can provide a link or openother issue where you provide some code so I can have a look at it.

**Comment by andrew-kennedy** [comment]
Well one example where a question arises is [here](https://github.com/hpicgs/glexamples/blob/master/source/openglexample/OpenGLExample.cpp#L118) where you bind a VAO using globjects but then draw the triangles using a standard openGL call, `glDrawArrays`. Why not use the `vao->drawArrays()` method here? Is it possible to accomplish this same pipeline using only globjects calls?

**Comment by andrew-kennedy** [comment]
also, here's some code I have questions about:
```
    auto g_vao = make_ref<VertexArray>();
    auto g_vertexBuffer = make_ref<Buffer>();
    g_vertexBuffer->setData(g_vertices, GL_STATIC_DRAW);
    auto g_colorBuffer = make_ref<Buffer>();
    g_colorBuffer->setData(g_colors, GL_STATIC_DRAW);

    auto binding = g_vao->binding(0);
    binding->setAttribute(0);
    binding->setBuffer(g_vertexBuffer, 0, sizeof(glm::vec3));
    binding->setFormat(3, GL_FLOAT);
    
    binding->setAttribute(1);
    binding->setBuffer(g_colorBuffer, 0, sizeof(glm::vec3));
    binding->setFormat(3, GL_FLOAT);
    
    g_vao->enable(0);
```

and the relevant vertex shader is:
```
#version 330 core
layout(location = 0) in vec3 vertPos;
layout(location = 1) in vec3 color;
uniform mat4 P;
uniform mat4 MV;

out vec3 Color;

void main()
{
    Color = color;
    gl_Position = P * MV * vec4(vertPos, 1.0);
}
```

Does this code make sense to initialize 2 buffers if `g_vertices` and `g_colors` are `std::vector<glm::vec3>`'s? (Ignoring the uniform matrices for now)
